### PR TITLE
HACS support: Build lovelace-floorplan as standalone file and more

### PR DESCRIPTION
**Pull request by exetico**
Will include both jQuery and YAML-handling in the build-process. 

I did try with Rollup without Browserify - but couldn't fix all the problems

HACS is supported now, if it's added a a custom repo. You can add it to the store too, if you like.

### CODE REVIEW

**Comment by kasperbjerby** [comment]
Could you include the long click version into this? :)
https://github.com/pkozul/lovelace-floorplan/pull/29

**Comment by exetico** [comment]
Damn, more ressources to come. Are you able to do it without the Longpres-lib, @kasperbjerby ? Else, I'll try and see how that part works out. I guess it will work the same way :-)!

**Comment by kasperbjerby** [comment]
The only part I need is longpress :) Basically I just want to be able to press and hold to get the more info box to open

**Comment by exetico** [comment]
Hi @kasperbjerby 

It's now added. 

I'm trying to get it to work with my configs, but it's not working.... 


Here's one example:
```
name: Office Spotlys
entities:
  - group.office
element: area.office
# groups: # Can be used to open group
#   - group.office
long_click:
  service: light.toggle
  data:
    entity_id: light.spotlys_kontor
  # data:
  #   card:
  #     entities:
  #       - entity: device_tracker.gm1913
  #     hours_to_show: 12
  #     type: map
  #   deviceID: this
  #   style:
  #     '--more-info-header-background': 'rgba(255, 159, 9, 1)'
  #   title: Chop chop
  # service: browser_mod.popup
action: null
  # service: light.toggle
  # data:
  #   entity_id: light.spotlys_kontor
states:
  - state: 'on'
    class: 'light-on'
  - state: 'off'
    class: 'light-off'
```
My config is a bit splitted up (to prevent it from getting too large). But my long_click are not working.

By clicking it "normally", it's showing the `group.office`-entities. 

I've also tried with the example in #29, but It's not working (I've installed browser mod, but it doesn't make any difference).

Could you kindly a few usefull examples? I'll put it in the readme, so people knows how to play with it ;-)

And oh. Kindly test it, and let me know if it's working as it should. There's no errors in the console, and the lib is integrated, as it should.

**Comment by exetico** [comment]
Oh, I added the longpress lib after floorplan. What a great plan... It's now corrected, and worked as it should.

Have you used the "preload" solution?

I would like to add some documentation about that, too:
```yaml
preload_cards:
     - map
     - gauge
preload_rows:
     - lock-entity-row
```

**Comment by exetico** [comment]
@pkozul : I've spotted that the following libs are not included with this version:
- `svg-pan-zoom.min.js`
- `fully-kiosk.js`

Currently loaded here:
https://github.com/pkozul/lovelace-floorplan/blob/40d8897e0bd6bd3fb1405fcff65530dd873b2daf/www/floorplan/lib/floorplan.js#L136-L148

It's included in the original [ha-floorplan](https://github.com/pkozul/ha-floorplan/tree/master/www/custom_ui/floorplan/lib), but I'm not sure, if it will work with the Lovelace-version? 

The Lovelace-version works without the normal ha-floorplan version, so I'm not sure if this part should be removed, of if I need to integrate that, too?

File: README.md [hunk 1]
```diff
@@ -58,9 +60,102 @@ You can then start adding floorplan cards to your Lovelace config. Under `entiti
...
+              element: area.office
+              entities:
+                - light.triplespot_office
+              ...
+```
+
+Credit(s): [Alfiegerner](https://github.com/Alfiegerner) - _[exetico](https://github.com/exetico) has provided a hot-fix for the logic(s). `image` will be used, if `image_mobile` are not defined._
+
+### Option: `pre_load`
+The `pre_load_ option are implemented to prevent "not found errors" for cards like hui-gauge. It's onclear if it's still required for things to work as expected. I'm currently [waiting for more details](https://github.com/pkozul/lovelace-floorplan/pull/29#issuecomment-731731289).
```
Comment (blackshadev) [inline README.md]: don't you mean gui-gauge?

try:
`It is still unclear if it is still required`
Don't try to contract to many things, it makes it harder to read. And I am not a native speaker, but I think one of those `it's` is wrong.

**Comment by blackshadev** [comment]
This is one massive PR sir! I don't think the maintainer even reads through these PRs anymore, since he didn't even bother with my PR. But if you continue to put all this great work in one massive PR it becomes unmaintainable for anyone to clearly what you did, why and to review it in a normal sense.  
So maybe... if you want to maintain this, fork it, thank to original maintainer for his great work, offer to merge it back or welcome him to your fork when/if he comes back. And let's get this thing up on HACS! I got some more idea's I want to put in, like template styling tags, to get your light icons the color they actually are!

In short: Great work, let's get this thing going!

**Comment by exetico** [comment]
Thanks!

I think you're right! But, Petar actually reached out by mail and via HA's Community, too. I'll wait for a reply, before doing anything else. _With that said, I didn't expect it :-) I'm just happy to hear form you,_ @pkozul .

I don't want to speak up for other persons, that's for sure. But Petar mentioned that a new version are currently "being builded", but I'm unsure about the current status of it. Maybe we'll hear more from I'm, in a matter of days :-)

File: README.md (hunk 1 above)
Comment (exetico) [reply README.md]: That's a good point. I found the option in #29, but couldn't really get the point of it.

Yesterday Alfiegerner provided an answer at https://github.com/pkozul/lovelace-floorplan/pull/29#issuecomment-731811522, so I'll try and look into that part.

**Comment by blackshadev** [comment]
> Thanks!
> 
> I think you're right! But, Petar actually reached out by mail and via HA's Community, too. I'll wait for a reply, before doing anything else. _With that said, I didn't expect it :-) I'm just happy to hear form you,_ @pkozul .
> 
> I don't want to speak up for other persons, that's for sure. But Petar mentioned that a new version are currently "being builded", but I'm unsure about the current status of it. Maybe we'll hear more from I'm, in a matter of days :-)

Ok, good to know. I haven't seen allot of activity and started to assume this project became.. abandon-ware as soon as I saw some activity I grabbed it as an opportunity to not let it be abandoned, I really like the project, use it every day, so I hope it will grow more feature rich. And if the activity starts flowing I am happy to contribute. But it isn't really motivating if you open PRs to not hear anything. That is why I didn't  bother to continue. Not the best of attitude (from me), but it is how it is. 

Anyway I will keep an close I on this thread!

**Comment by pkozul** [comment]
HI guys. Sorry about my absence on here. I had some momentum going on this project earlier this year, buth life got in the way (father passed away after a tough battle in July) and so it's good to see you guys are still using Floorplan. If you guys are happy to get involved, I can give you access to the repo.

As I mentioned to Tobias, I have a new version that is almost ready, which I am now preparing. Any changes from this current version we can retro-fit into the new one. The new one has been written in TypeScript, and no longer requires any external libraries (i.e. jQuery, YAML lb, etc.). I have built a HA entity state simulator, so Floorplan can be used in a minimal setup even without Home Assistant.

In any case, it's time to get this project back on track.

Cheers,
Petar

**Comment by blackshadev** [comment]
> HI guys. Sorry about my absence on here. I had some momentum going on this project earlier this year, buth life got in the way (father passed away after a tough battle in July) and so it's good to see you guys are still using Floorplan. If you guys are happy to get involved, I can give you access to the repo.
> 
> As I mentioned to Tobias, I have a new version that is almost ready, which I am now preparing. Any changes from this current version we can retro-fit into the new one. The new one has been written in TypeScript, and no longer requires any external libraries (i.e. jQuery, YAML lb, etc.). I have built a HA entity state simulator, so Floorplan can be used in a minimal setup even without Home Assistant.
> 
> In any case, it's time to get this project back on track.
> 
> Cheers,
> Petar

Good to hear from you. I hope you are doing well, since you went through quite allot. 
I am eager to see your new version and I will happily retro fit my tiny pr and do some more work. I love typescript, so yeeh for that. And yeeh for your awesome project. It is literally the gateway to every smart device in my house

**Comment by exetico** [comment]
@blackshadev - Could you try and explain what the pre_load are used to, in floorplan? I've not found a single case, where my cards are not displayed correcly.

I did try and ping you on HA's Community, but I guess you're using another username there, right?

Fell free to answer on HA instead, to prevent a real mess in this PR (_Well, it's already happening_)

> Are you able to provide a few works about the pre_load solution? I've not seen a single error, but maybe I've just missed the point of the pre_load solution.
> 
> https://github.com/pkozul/lovelace-floorplan/pull/43
> 
> I've looked in the HA thread related to this, too - but couldn't find anything relevant.
> 
> Could I ask you to find an example, where I can see the difference, with- and without pre_load in use? :-) 
> 
> I found it a bit better to have this chat here - I hope that's OK!
> 

And oh, I totally understand why you pointed me in the direction you did. I would have done the same, if I've waited that long. That's for sure! 👍

**Comment by blackshadev** [comment]
> @blackshadev - Could you try and explain what the pre_load are used to, in floorplan? I've not found a single case, where my cards are not displayed correcly.
> 
> I did try and ping you on HA's Community, but I guess you're using another username there, right?
> 
> Fell free to answer on HA instead, to prevent a real mess in this PR (_Well, it's already happening_)
> 
> > Are you able to provide a few works about the pre_load solution? I've not seen a single error, but maybe I've just missed the point of the pre_load solution.
> > #43
> > I've looked in the HA thread related to this, too - but couldn't find anything relevant.
> > Could I ask you to find an example, where I can see the difference, with- and without pre_load in use? :-)
> > I found it a bit better to have this chat here - I hope that's OK!
> 
> And oh, I totally understand why you pointed me in the direction you did. I would have done the same, if I've waited that long. That's for sure! 👍

I don't think I have an HA community account yet.
But I don't know anything about the pre_load. Didn't make it, didn't use it. Are you confusing me with someone else?

**Comment by exetico** [comment]
> I don't think I have an HA community account yet.
> But I don't know anything about the pre_load. Didn't make it, didn't use it. Are you confusing me with someone else?

No, [you commented on that part](https://github.com/pkozul/lovelace-floorplan/pull/43#pullrequestreview-536748620). And therefore I thought you've used it :-)

**Comment by exetico** [comment]
We're ready for version 1.0.0 🥳

**Comment by blackshadev** [comment]
> > I don't think I have an HA community account yet.
> > But I don't know anything about the pre_load. Didn't make it, didn't use it. Are you confusing me with someone else?
> 
> No, [you commented on that part](https://github.com/pkozul/lovelace-floorplan/pull/43#pullrequestreview-536748620). And therefore I thought you've used it :-)

Nope. I commented on your amazing work. Havent used it, yet.
